pkg/controller/telemetry/metric.go

```diff
@@ -235,29 +395,47 @@ func buildPrincipal(workload *workloadapi.Workload) string {
 	return "-"
 }
 
-func buildMetricsToPrometheus(data requestMetric, labels commonTrafficLabels) {
-	commonLabels := commonTrafficLabels2map(&labels)
-	tcpConnectionOpened.With(commonLabels).Set(float64(data.connectionOpened))
-	tcpConnectionClosed.With(commonLabels).Set(float64(data.connectionClosed))
-	tcpReceivedBytes.With(commonLabels).Set(float64(data.receivedBytes))
-	tcpSentBytes.With(commonLabels).Set(float64(data.sentBytes))
+func buildWorkloadMetricsToPrometheus(data requestMetric, labels workloadMetricLabels) {
+	commonLabels := struct2map(labels)
+	tcpConnectionOpenedInWorkload.With(commonLabels).Add(float64(1))
+	tcpConnectionClosedInWorkload.With(commonLabels).Add(float64(1))
+	if data.success != uint32(1) {
+		tcpConnectionFailedInWorkload.With(commonLabels).Add(float64(1))
+	}
+	tcpReceivedBytesInWorkload.With(commonLabels).Add(float64(data.receivedBytes))
+	tcpSentBytesInWorkload.With(commonLabels).Add(float64(data.sentBytes))
+}
+
+func buildServiceMetricsToPrometheus(data requestMetric, labels serviceMetricLabels) {
+	commonLabels := struct2map(labels)
+	tcpConnectionOpenedInService.With(commonLabels).Add(float64(1))
+	tcpConnectionClosedInService.With(commonLabels).Add(float64(1))
```

Comment:
why should tcpConnectionOpenedInService always equal to tcpConnectionClosedInService

How could user know how many connection is open now